synapse/replication/tcp/handler.py

```diff
@@ -71,7 +83,41 @@ def __init__(self, hs):
         self.factory = None  # type: Optional[ReplicationClientFactory]
 
         # The currently connected connections.
-        self.connections = []
+        self.connections = []  # type: List[Any]
+
+        LaterGauge(
+            "synapse_replication_tcp_resource_total_connections",
+            "",
+            [],
+            lambda: len(self.connections),
+        )
+
+        LaterGauge(
+            "synapse_replication_tcp_resource_connections_per_stream",
+            "",
+            ["stream_name"],
+            lambda: {
+                (stream_name,): len(
+                    [
+                        conn
+                        for conn in self.connections
+                        if stream_name in conn.replication_streams
+                    ]
+                )
+                for stream_name in self.streams
+            },
+        )
```

Comment:
I spent ages trying to figure out what was going on here before realising that this is reintroducing stuff that got removed in #7195. AFAICT this reintroduces the problem that that PR tried to fix.

please can you update this branch, I guess.